### ISSUE TITLE
Is this a typo on line 75 of nni/nas/pytorch/mutables.py?

### ISSUE THREAD

**Issue by Davidxswang** [microsoft/nni#3515]
While I was reading the source code, I wondered whether this is a typo, as shown below:

nni/nas/pytorch/mutables.py, line 75
```
70    @property
71    def name(self):
72        """
73        After the search space is parsed, it will be the module name of the mutable.
74        """
75        return self._name if hasattr(self, "_name") else "_key"
```
Should `"_key"` be `self._key` because this method is supposed to return a key/name of a mutable?

**Comment by ultmaster** [comment]
I doubt that you are right. I think it runs without a problem because this branch is not actually needed.

Submit a PR to fix it if you want.

**Comment by ankitaggarwal23** [comment]
@Davidxswang Can I work on this issue?

**Comment by Davidxswang** [comment]
> @Davidxswang Can I work on this issue?

Sure! It would be much appreciated! Thank you!

**Comment by ankitaggarwal23** [comment]
@Davidxswang It would be great If you could assign me this issue. As I'm a first time contributor.

**Comment by Davidxswang** [comment]
> @Davidxswang It would be great If you could assign me this issue. As I'm a first time contributor. I have a small doubt just correct me if I'm wrong in any point.
> 
> I just want to replace `"_key" ` along with quotes and change it `else self.key` Just like that,
> 
> ```
>  @property
> 71    def name(self):
> 72        """
> 73        After the search space is parsed, it will be the module name of the mutable.
> 74        """
> 75        return self._name if hasattr(self, "_name") else self.key
> ```

Hi @ankitaggarwal23 I think you are right, `self.key` or `self._key` would work. 
Sorry, I am not in this project so I cannot assign issues to anyone. I guess in order to contribute to this project, you have to fork the repo and fix the bug and then submit a PR. After that, whoever in the project will approve your PR and merge it into the main branch.

**Comment by ankitaggarwal23** [comment]
> > @Davidxswang It would be great If you could assign me this issue. As I'm a first time contributor. I have a small doubt just correct me if I'm wrong in any point.
> > I just want to replace `"_key" ` along with quotes and change it `else self.key` Just like that,
> > ```
> >  @property
> > 71    def name(self):
> > 72        """
> > 73        After the search space is parsed, it will be the module name of the mutable.
> > 74        """
> > 75        return self._name if hasattr(self, "_name") else self.key
> > ```
> 
> Hi @ankitaggarwal23 I think you are right, `self.key` or `self._key` would work.
> Sorry, I am not in this project so I cannot assign issues to anyone. I guess in order to contribute to this project, you have to fork the repo and fix the bug and then submit a PR. After that, whoever in the project will approve your PR and merge it into the main branch.

@Davidxswang Sure, Thank you for guiding!

**Comment by Davidxswang** [comment]
> > > @Davidxswang It would be great If you could assign me this issue. As I'm a first time contributor. I have a small doubt just correct me if I'm wrong in any point.
> > > I just want to replace `"_key" ` along with quotes and change it `else self.key` Just like that,
> > > ```
> > >  @property
> > > 71    def name(self):
> > > 72        """
> > > 73        After the search space is parsed, it will be the module name of the mutable.
> > > 74        """
> > > 75        return self._name if hasattr(self, "_name") else self.key
> > > ```
> > 
> > 
> > Hi @ankitaggarwal23 I think you are right, `self.key` or `self._key` would work.
> > Sorry, I am not in this project so I cannot assign issues to anyone. I guess in order to contribute to this project, you have to fork the repo and fix the bug and then submit a PR. After that, whoever in the project will approve your PR and merge it into the main branch.
> 
> @Davidxswang Sure, Thank you for guiding!

@ankitaggarwal23 You are very welcome!

**Comment by Davidxswang** [comment]
I'll close it since this has been fixed in version 2.2.
